subgraphs/uniswap-forks/src/common/masterchef/helpers.ts

```diff
@@ -42,17 +42,26 @@ export function getOrCreateMasterChef(
 ): _MasterChef {
   let masterChef = _MasterChef.load(masterChefType);
 
-  if (!masterChef) {
-    masterChef = new _MasterChef(masterChefType);
+  if (masterChef) {
+    if (masterChef.address) {
```

Comment:
Why do it like this and not just `if (masterchef)`

Reply:
`if(masterchef)` doesn't guarantee that the `address` field is set.

The idea is:
if masterchef exists and the address is set, return as is
if it exists but the address is not set, set it and save, then return
if it doesn't exist, create

Reply:
So is there an instance where you would be creating a masterchef entity but not add the address? If not, then isn't it safe to assume that the address is there?

Reply:
For subgraphs that are already deployed, that already have the masterchef instance created but it happened before the address was added.

Shouldn't be a problem for subgraphs that sync from scratch, but it will for the ones that are grafted from the old schema

Reply:
Oh I get. Thanks for explaining!